Update README.md to describe use of mpirc executble as a debuger shim layer (#27)
openpmix/mpir-to-pmix-guide: README.md

Note that Attach Mode assumes a Proxy Mode launch at this time. It may not work with the Non-Proxy mode.

## Using the MPIR Shim Module With Debuggers

The MPIR shim module can be used with debuggers that are debugging applications in launch mode or attach mode. Both modes will be demonstrated using
gdb and Open MPI below. Assume the MPIR shim module has been built and installed in /u/test/mpir.

### Using the MPIR Shim Module for Debugging in Launch Mode

> Use the debugger to launch the MPIR shim module

Reviewer: ```suggestion
Use the debugger to launch the MPIR Shim tool
```

~~~
gdb /u/test/mpir/bin/mpirc
~~~

> Set a breakpoint at MPIR_Breakpoint. This breakpoint will be hit once the mpirun process has built the MPI process table and it is available

Reviewer: ```suggestion
Set a breakpoint at ``MPIR_Breakpoint``. This breakpoint will be hit once the ``mpirun`` process has built the MPI process table and it is available
```

for use by a debugger. 

> Once the breakpoint has been set, launch mpirun and the application under control of the MPIR shim module. 

Reviewer: ```suggestion
Once the breakpoint has been set, have gdb run the ``mpirun`` launcher, which starts the application under the control of the MPIR shim module. 
```

> The operands to the gdb run command are as if the mpirun command was issued at a shell prompt. Any mpirun options are allowed as well as any parameters for the application program.

Reviewer: ```suggestion
The operands to the gdb ``run`` command are as if the ``mpirun`` command was issued at a shell prompt. Any ``mpirun`` options are allowed as well as any parameters for the application program.
```

```
(gdb) b MPIR_Breakpoint
(gdb) run mpirun -n 2 /u/test/mpi/bin/hello
```

> The launch will run until the MPIR process table in built. When the breakpoint at MPIR_Breakpoint is hit, you can display MPIR process table information.

Reviewer: ```suggestion
The launch will run until the MPIR process table in built. When the breakpoint at ``MPIR_Breakpoint`` is hit, you can display MPIR process table information.
```

```
Thread 1 "mpirc" hit Breakpoint 1, MPIR_Breakpoint () at mpirshim.c:219
219         MPIR_SHIM_DEBUG_ENTER("");
Missing separate debuginfos, use: yum debuginfo-install libevent-2.1.8-5.el8.ppc64le libgcc-8.4.1-1.el8.ppc64le nvidia-driver-cuda-libs-450.142.00-1.el8.ppc64le openssl-libs-1.1.1g-15.el8_3.ppc64le sssd-client-2.4.0-9.el8_4.2.ppc64le zlib-1.2.11-17.el8.ppc64le
(gdb) p MPIR_proctable_size
$1 = 2
(gdb) p MPIR_proctable[0]
$2 = {
  host_name = 0x20002400a0f0 "c656f7n05",
  executable_name = 0x100310e0 "/u/test/mpi/bin/hello",
  pid = 1501610
}
(gdb) p MPIR_proctable[1]
$3 = {
  host_name = 0x100310c0 "c656f7n05",
  executable_name = 0x100da940 "/u/test/mpi/bin/hello",
  pid = 1501611
}
```

Execution of mpirun and the target application continues after you enter the gdb **continue** command.

### Using the MPIR Shim Module for Debugging in Attach Mode

> Launch the application program and note the process id (pid) of the mpirun process. The application should run 
> long enough that you can start a debugger and attach to the mpirun process.

Reviewer: ```suggestion
Launch the application program and note the process ID (PID) of the ``mpirun`` process. The application should run 
long enough that you can start a debugger and attach to the ``mpirun`` process.
```

```
c656f7n05:test> mpirun -n 2 hello &
[1]     1501960

```

Invoke gdb to run the MPIR shim module

```
gdb /u/test/mpir/bin/mpirc
```

> Set a breakpoint at MPIR_Breakpoint then run the shim module, specifying the pid of the mpirun process.

Reviewer: ```suggestion
Set a breakpoint at ``MPIR_Breakpoint`` then run the shim module, specifying the PID of the ``mpirun`` process.
```

```
(gdb) b MPIR_Breakpoint
(gdb) run mpirun -c 1501960
```

> The launch will run until the MPIR process table in built. When the breakpoint at MPIR_Breakpoint is hit, you can display MPIR process table information as shown in the description of the mpirun launch case above.

Reviewer: ```suggestion
The launch will run until the MPIR process table in built. When the breakpoint at ``MPIR_Breakpoint`` is hit, you can display MPIR process table information as shown in the description of the ``mpirun`` launch case above.
```
